openshift/serverless.bash

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env bash
+
+function ensure_serverless_installed {
+  logger.info 'Check if Serverless is installed'
+  local prev=${1:-false}
+  if oc get knativeserving.operator.knative.dev knative-serving -n "${SERVING_NAMESPACE}" >/dev/null 2>&1 && \
+     oc get knativeeventing.operator.knative.dev knative-eventing -n "${EVENTING_NAMESPACE}" >/dev/null 2>&1
+  then
+    logger.success 'Serverless is already installed.'
+    return 0
+  fi
+
+  # Deploy config-logging configmap before running serving-opreator pod.
+  # Otherwise, we cannot change log level by configmap.
+  enable_debug_log
+
+  if [[ $prev == "true" ]]; then
+    install_serverless_previous
+  else
+    install_serverless_latest
+  fi
+}
+
+function install_serverless_previous {
+  local rootdir
+  rootdir="$(dirname "$(dirname "$(dirname "$(realpath "${BASH_SOURCE[0]}")")")")"
+
+  # Remove installplan from previous installations, leaving this would make the operator
+  # upgrade to the latest version immediately
+  remove_installplan "$CURRENT_CSV"
+
+  deploy_serverless_operator "$PREVIOUS_CSV"  || return $?
+  deploy_knativeserving_cr || return $?
+}
+
+function remove_installplan {
+  local install_plan
+  install_plan=$(find_install_plan $1)
+  if [[ -n $install_plan ]]; then
+    oc delete "$install_plan" -n "${OPERATORS_NAMESPACE}"
+  fi
+}
+
+function install_serverless_latest {
+  deploy_serverless_operator_latest || return $?
+  deploy_knativeserving_cr || return $?
+}
+
+function deploy_serverless_operator_latest {
+  local rootdir
+  rootdir="$(dirname "$(dirname "$(dirname "$(realpath "${BASH_SOURCE[0]}")")")")"
```

Comment:
why is this needed here? I don't see `rootdir` referenced anywhere in the subsequent calls?

Reply:
This is ported from master/release next, in this commit: 
https://github.com/openshift/knative-eventing/commit/22b19f52dd4c3e30e18245e62671c96d9896a9ff

what it does is:
* it installs the serverless operator, with a patch `serverless.bah`, to make sure we only install `Knatrive Serving`.

Reply:
here we bascially "patch" the operator:
https://github.com/openshift/knative-eventing/commit/22b19f52dd4c3e30e18245e62671c96d9896a9ff#diff-cd758740b9c6da1c23340da7084242feR71

unforntately the `hack/install.sh` has no "flag" or so to say: I just want serving, hence my patch/hack